plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/jmx/VersionDetector.java

```diff
@@ -0,0 +1,46 @@
+package org.jboss.tools.openshift.core.jmx;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.wst.server.core.IServer;
+import org.jboss.tools.openshift.core.server.OpenShiftServerUtils;
+import org.jboss.tools.openshift.internal.core.util.ResourceUtils;
+
+import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.images.DockerImageURI;
+import com.openshift.restclient.model.IBuildConfig;
+import com.openshift.restclient.model.IProject;
+import com.openshift.restclient.model.IService;
+
+public class VersionDetector {
+	private static final Map<String, VersionKey> knowImages= buildVersionMap();
+	
+	private IServer server;
+	public VersionDetector(IServer server) {
+		this.server= server;
+	}
+	
+	private static Map<String, VersionKey> buildVersionMap() {
+		Map<String, VersionKey> result= new HashMap<String, VersionKey>();
+		result.put("jboss-eap64-openshift", new VersionKey(ServerType.EAP, "6.4.0"));	

```

Comment:
These strings don't mean anything beyond their literal value and they should not be referenced anywhere outside this method so I don't think a constant is appropriate. Unless we want to have a registry of well known image names, but then it should not live hidden inside this class.